src/libraries/System.Text.Json/gen/TypeGenerationSpec.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.Json.Serialization;
+using System.Text.Json.SourceGeneration.Reflection;
+
+namespace System.Text.Json.SourceGeneration
+{
+    [DebuggerDisplay("Type={Type}, ClassType={ClassType}")]
+    internal class TypeGenerationSpec
+    {
+        private bool _hasBeenInitialized;
+
+        /// <summary>
+        /// Fully qualified assembly name, prefaced with "global::", e.g. global::System.Numerics.BigInteger.
+        /// </summary>
+        public string TypeRef { get; private set; }
+
+        /// <summary>
+        /// The name of the public JsonTypeInfo<T> property for this type on the generated context class.
+        /// For example, if the context class is named MyJsonContext, and the value of this property is JsonMessage;
+        /// then users will call MyJsonContext.JsonMessage to access generated metadata for the type.
+        /// </summary>
+        public string TypeInfoPropertyName { get; set; }
+
+        public bool GenerateMetadata { get; set; } = true;
+
+        private bool? _generateSerializationLogic;
+        public bool GenerateSerializationLogic
+        {
+            get => _generateSerializationLogic ??= FastPathIsSupported();
+            set => _generateSerializationLogic = value;
```

Comment:
When would the setter be called?

Reply:
The setter is called here when we determine what to generate given user configuration:

https://github.com/dotnet/runtime/blob/ccb37060263ba7d0c2dff22e9ae5e2f0d564f9c2/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L284-L296